### ISSUE TITLE
`DateTime.fromFormat('31', 'dd').isValid` result failing based on the current month

### ISSUE THREAD

**Issue by dennis-zinzi** [moment/luxon#1195]
**Describe the bug**
Checking whether a day-only DateTime value (formats `d` or `dd`) is valid via the `.isValid` method produced inconsistent results based on the current month of the year.

**To Reproduce**
Snippet to run:
```js
const { DateTime } = require("luxon");

const isDateInFormat = DateTime.fromFormat(
  '31',
  'dd',
  { locale: 'GB' },
).isValid;

if (!isDateInFormat) {
  console.error(
    'Date not in given format "dd"',
  );
} else {
  console.log('Valid Date');
}
```

If running this code in a 30-day month like April, the resulting log is: `Error: Date not in given format "d"`

Whilst if you call the same snippet in a 31-day month like May, the resulting log is: `Info: Valid Date`

**Actual vs Expected behavior**
Expected: Validating for a valid Day value would be irrelevant of the month in which the code is running, or at least have a flag for the validation to be month/year agnostic (year agnostic for days like Feb 29th)

Actual: The result of validating a Day-only value (format `d` or `dd`) is based on the current month (`31` valid in May, but invalid in April) despite not needing such context

**Desktop (please complete the following information):**

- OS: "macOS/Linux/AWS Lambda"
- Browser: "Node v14.17.0"
- Luxon version: "1.26.0"
- Your timezone: "Europe/London"

**Additional context**
Think all info has been provided, but feel free to ask for more details if such are needed.

Many thanks for looking into this.

**Comment by icambron** [comment]
That's right. Luxon DateTimes are specific times; it doesn't have a way to represent the 31st day of an unspecified month. DateTime.fromFormat's job is to return a DateTime, which knows when it is to the millisecond.

You are supplying limited information to the parser, and it's defaulting the missing pieces. The elements bigger than what you specified (year, month) default to their now() values, and the smaller ones (hours, minutes, seconds) default to 0. It puts all that together and asks, "is this valid?"
